lib/streamlit/elements/exception.py

```diff
@@ -56,29 +56,45 @@ def exception(self, exception: BaseException) -> DeltaGenerator:
         >>> st.exception(e)
 
         """
-        exception_proto = ExceptionProto()
-        marshall(exception_proto, exception)
-        return self.dg._enqueue("exception", exception_proto)
+        return _send_proto(self.dg, exception)
 
     @property
     def dg(self) -> DeltaGenerator:
         """Get our DeltaGenerator."""
         return cast("DeltaGenerator", self)
 
 
-def marshall(exception_proto: ExceptionProto, exception: BaseException) -> None:
+# TODO(lawilby): confirm whether we want to track metrics here with lukasmasuch.
+@gather_metrics("exception")
+def _send_proto(
```

Comment:
nit: I would call this `_exception` which would be aligned with other files where we do something similar for the public-facing vs. non-public facing API (e.g. in button.py)